internal/db/code.go

```diff
@@ -56,61 +62,77 @@ var boolValidator = Code{
 	ID:       uuid.MustParse("8e806967-c462-47af-8756-48674537a909"),
 	Name:     "boolValidator",
 	Runtime:  Golang,
-	Function: Validator,
+	Function: datatypes.BoolFromJSON,
 }
 
 var intValidator = Code{
 	ID:       uuid.MustParse("a1cf1c16-040d-482c-92ae-92d59dbad46c"),
 	Name:     "intValidator",
 	Runtime:  Golang,
-	Function: Validator,
+	Function: datatypes.IntFromJSON,
 }
 
 var enumValidator = Code{
 	ID:       uuid.MustParse("5c3b9da9-c592-41da-b6e2-8c8dd97186c3"),
 	Name:     "enumValidator",
 	Runtime:  Golang,
-	Function: Validator,
+	Function: datatypes.EnumFromJSON,
 }
 
 var stringValidator = Code{
 	ID:       uuid.MustParse("aaeccd14-e69f-4561-91ef-5a8a75b0b498"),
 	Name:     "stringValidator",
 	Runtime:  Golang,
-	Function: Validator,
+	Function: datatypes.StringFromJSON,
 }
 
 var textValidator = Code{
```

Comment:
these validators belong in their own file

Reply:
Metamodel?